lib/puppet/transaction/report.rb

```diff
@@ -158,6 +168,21 @@ def compute_status(resource_metrics, change_metric)
   end
 
   # @api private
+  def has_noop_change_events?(resource)
+    resource.events.any? { |event| (event.status == 'noop' && event.name == :content_changed) }
```

Comment:
There are probably more change options:

```
    events:
    - !ruby/object:Puppet::Transaction::Event
      audited: false
      property: content
      previous_value: "{md5}4e4cdf645e477bdcee0810dccc9c538d"
      desired_value: "{md5}81e4d91298307865cae873f56768b2f4"
      historical_value:
      message: current_value {md5}4e4cdf645e477bdcee0810dccc9c538d, should be {md5}81e4d91298307865cae873f56768b2f4
        (noop)
      name: :content_changed
      status: noop
      time: 2016-06-07 05:37:14.165072743 -07:00
```

Maybe, it should compare `previous_value` and `desired_value` instead.